### PR TITLE
2023 cleanup, separating conf & content, add friends

### PR DESCRIPTION
**Pull request by rlc4**
Did a bunch of cleanup in prep for 2024, breaking the 'few changes per commit" rule

- Separatred all Content and Config
   * index.html, robots.txt, and favicon.ico were relocated out of the root directory
   * updated links to reference the new locations

- 2023 cleanup
  - moved a lot of content into _pages/2023 to make it easy to find for 2024 re-use of content
  - updated nav to be minimal between events content
  - Updated the sponsors.yaml with the current offerings all as empty spots
  - updated logo and color to default (hope I got the right color here)
- Set up the "Friends" page, including the related YAML.  not linked in navigation
- fixed the subcategories a href to it'll open a new window

### CODE REVIEW

**Comment by benjamin-hering** [comment]
LGTM. Runs fine locally without any glaring issues.

Screenshot 2023-08-28 at 5 31 37 PM

**Review by benjamin-hering — APPROVED** [review]
👍

File: _data/friends.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- name: BSidesSF
+  link: https://bsidessf.org/
+- name: OWASP
+  link: https://owasp.org/
+- name: EFF
+  link: https://eff.org/
+- name: BSidesSF
+  link: https://linkedin.com/in/bsidessf/
+- name: Hiro Protagonist
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/friends.yml:9]: This is fake data to set up the "Friends of BSidesSF" page(s).  Right now it does render the page but it's not linked into the NAV at all.

File: _data/friends.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+- name: BSidesSF
+  link: https://bsidessf.org/
+- name: OWASP
+  link: https://owasp.org/
+- name: EFF
+  link: https://eff.org/
+- name: BSidesSF
+  link: https://linkedin.com/in/bsidessf/
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/friends.yml:8]: Once we have a "Friend of..." signed up, we'll empty this yaml and fill with Friends.

File: _data/friends.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+- name: BSidesSF
+  link: https://bsidessf.org/
+- name: OWASP
+  link: https://owasp.org/
+- name: EFF
+  link: https://eff.org/
+- name: BSidesSF
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/friends.yml:7]: See abvoe

File: _data/friends.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+  link: https://bsidessf.org/
+- name: OWASP
+  link: https://owasp.org/
+- name: EFF
+  link: https://eff.org/
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/friends.yml:6]: See above

File: _data/nav.yml [hunk 5]
```diff
@@ -1,6 +1,12 @@
 - title: "Home"
   href: "/"
 
+#- title: "Sponsors"
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/nav.yml:4]: Adding a commit to bring this into NAV, will also update the link for the sponsor package.

File: _data/sponsors.yml [hunk 6]
```diff
@@ -1,207 +1,189 @@
-sponsorship_kit_url: "https://drive.google.com/open?id=1r4R8VWQRyBTQL7b_UoDhcLFjAiDePk9O"
+#
+# level - creates a section with this key as the header
+#       - used for categories with more than one sponsorship availabled
+# type - puts a header on the single sponsor
+#      - best for single sponsor items
+# break - puts in a row break
+#
+sponsorship_kit_url: "GOOGLE DOCS LINK"
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/sponsors.yml:8]: Thanks, fixing with an additional commit

File: _pages/graphic-recordings.html
[the remark this answers is not in the document]
Comment (rlc4) [reply _pages/graphic-recordings.html:1]: It's functional for me, but it coiuld definitly use some beautification

File: _includes/head.html [hunk 7]
```diff
@@ -18,7 +18,7 @@
     
   <link href="{{ site.baseurl }}/css/main.css" media="screen, projection" rel="stylesheet" type="text/css">
   <link rel="canonical" href="{{ site.url }}{{ site.baseurl }}{{ page.url | remove:'index.html' }}">
-  <link href="{{ site.baseurl }}/favicon.ico" rel="shortcut icon" type="image/x-icon">
+  <link href="{{ site.baseurl }}/images/favicon.ico" rel="shortcut icon" type="image/x-icon">
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _includes/head.html:21]: I think the HTML4.01 standard might be different:
- https://en.wikipedia.org/wiki/Favicon 
W3Schools example shows favicon in images subdirectory
- https://www.w3schools.com/html/html_favicon.asp

File: _pages/about/index.md
[the remark this answers is not in the document]
Comment (rlc4) [reply _pages/about/index.md:1]: If you're referring to moving a bunch of files to 2023 -- I did that so we would have previous year content easily available, but not have the files in the current directory so it was clear if they had been updated or not.   Once we have the 2024 versions I expected the 2023 subdir to go away.

If this isn't what "this" refers to, please let me know.   Github has this comment in a weird spot unlinked to any file, and vscode has it associated with the about page.

File: _pages/robots.txt
[the remark this answers is not in the document]
Comment (rlc4) [reply _pages/robots.txt:1]: I should have put a few of these changes into a new PR, this being one of them.
The idea here is to keep content and control separated.
(Favicon and index.md moving are part of the same thought process.)

File: _data/nav.yml [hunk 8]
```diff
@@ -1,6 +1,12 @@
...
+#  href: "/sponsors"
+
+#- title: "Friends"
```
[the remark this answers is not in the document]
Comment (rlc4) [reply _data/nav.yml:7]: Further down moves it past the "About" navigation, I thought it was more similar to Sponsors than being after About, which is generic about BSidesSF, has history, policies, etc in it

File: graphic-recordings.html
[the remark this answers is not in the document]
Comment (rlc4) [reply graphic-recordings.html:1]: As far as I can tell, it does.   If it's not working let's get it fixed.

File: index.html
[the remark this answers is not in the document]
Comment (rlc4) [reply index.html:1]: I could use some help here. 

I tried to go back to the relevant commit and use --amend, cannot seem to change this properly to show the file moved and was modified after moving.

Attempted
`
# commit I want to amend
git checkout 2819f511ff7a428d0d40d3c3328cfac912ad4e53  
# to get the prev version of index.html
git checkout 8d359efe3e878cba575cf4f6314baf9633b445c3 index.html   
# stash the edited index.html so we can put it back later
cp _pages/index.html _pages/index.html.safe  
git mv index.html _pages/index.html
cp _pages/index.html.safe _pages/index.html
git add _pages/index.html
# this shows no changes to be committed
git status
# blindly try to amend the commit, this appears to do nothing
git commit --amend 
`

**Comment by rlc4** [comment]
I'm going to close the PR in favor of a better set of individual PRs with clearer atomic changes.

**Pull request closed without merging**